### PR TITLE
Fix/select ticket redesign

### PR DESCRIPTION
**Pull request by asogwachiuba**
This contains:
Ticket user flow correction
Home Stacked widget flow fix
Event Item null error fix

### CODE REVIEW

File: lib/features/events/views/event_detail/event_details_view.dart [hunk 1]
```diff
@@ -719,164 +751,156 @@ class EventDetailsViewState extends State<EventDetailsView> {
...
+                              }
+                            }
+                          }))),
+              SizedBox(width: 20.w),
+              GestureDetector(
+                  onTap: () {
+                    final msg = "Added event to calendar";
+                    final Event event = Event(
+                      title: currentEvent!.title!,
+                      description: currentEvent!.description!,
```
Comment (divincoder) [inline lib/features/events/views/event_detail/event_details_view.dart:885]: I said you should avoid using !. What if sth is null, description title and boom the app throws error.
Change to ?. then at the end do ?? "". Empty string as default incase.

File: lib/features/events/views/event_detail/event_details_view.dart [hunk 2]
```diff
@@ -719,164 +751,156 @@ class EventDetailsViewState extends State<EventDetailsView> {
...
+                      location:
+                          '${currentEvent!.location?.address ?? ""}, ${currentEvent!.location?.name ?? ""}',
+                      startDate: DateTime.tryParse(currentEvent!.startTime!)!
+                          .toLocal(),
+                      endDate:
+                          DateTime.tryParse(currentEvent!.endTime!)!.toLocal(),
```
Comment (divincoder) [inline lib/features/events/views/event_detail/event_details_view.dart:891]: Same here. Stop using !. It is dangerous

**Review by divincoder — CHANGES_REQUESTED** [review]

File: lib/features/events/views/event_detail/event_details_view.dart (hunk 1 above)
Comment (asogwachiuba) [reply lib/features/events/views/event_detail/event_details_view.dart]: I have fixed it. This was written last year i think or early this year so i didnt see it recently. Will be more cautious about it

File: lib/features/events/views/event_detail/event_details_view.dart (hunk 2 above)
Comment (asogwachiuba) [reply lib/features/events/views/event_detail/event_details_view.dart]: Fixed too

**Comment by asogwachiuba** [comment]
I have resolved the merge conflict. Came from the recent changes to the develop branches as i previously merged it with develop branch before making a pull request.

**Pull request merged**